src/hotspot/cpu/riscv/templateTable_riscv.cpp

```diff
@@ -2215,13 +2215,83 @@ void TemplateTable::load_field_cp_cache_entry(Register obj,
   }
 }
 
+// The xmethod register is input and overwritten to be the adapter method for the
+// indy call. Return address (ra) is set to the return address for the adapter and
+// an appendix may be pushed to the stack. Registers x10-x13 are clobbered.
+void TemplateTable::load_invokedynamic_entry(Register method) {
+  // setup registers
+  const Register appendix = x10;
+  const Register cache = x12;
+  const Register index = x13;
+  assert_different_registers(method, appendix, cache, index, xcpool);
+
+  __ save_bcp();
+
+  Label resolved;
+
+  __ load_resolved_indy_entry(cache, index);
+  __ membar(MacroAssembler::AnyAny);
```

Comment:
Why is the AnyAny barrier used here？

Reply:
Hi @sunny868, I'm working on removing these unnecessary barriers. RISC-V port uses more conservative barriers like this for some reasons (e.g.: [1][2][3]),  we can just remove them. 

1. https://github.com/openjdk/jdk/blob/36ec05d52a79185d8c6669713fd17933128c032a/src/hotspot/cpu/riscv/templateTable_riscv.cpp#L3438-L3443
2. https://github.com/openjdk/jdk/blob/36ec05d52a79185d8c6669713fd17933128c032a/src/hotspot/cpu/riscv/templateTable_riscv.cpp#L3558-L3563
3. https://github.com/openjdk/jdk/blob/36ec05d52a79185d8c6669713fd17933128c032a/src/hotspot/cpu/riscv/templateTable_riscv.cpp#L3614-L3619